internals/secrethub/generate.go

```diff
@@ -74,7 +85,56 @@ func (cmd *GenerateSecretCommand) before() error {
 		return err
 	}
 
-	cmd.generator = randchar.NewGenerator(useSymbols)
+	charset := randchar.Alphanumeric
+	if useSymbols {
+		symbols, found := randchar.CharsetByName("symbols")
+		if found {
+			charset = charset.Add(symbols)
+		} else {
+			return ErrCouldNotFindCharSet("symbols")
+		}
+	}
+
+	var includedCharsets []randchar.Charset
+	for _, charsetName := range cmd.includes {
+		charsetToInclude, found := randchar.CharsetByName(charsetName)
+		if !found {
+			return ErrCouldNotFindCharSet(charsetName)
+		}
+		charset = charset.Add(charsetToInclude)
+		includedCharsets = append(includedCharsets, charsetToInclude)
+	}
+
+	for _, charsetName := range cmd.excludes {
+		charsetToExclude, found := randchar.CharsetByName(charsetName)
+		if !found {
+			return ErrCouldNotFindCharSet(charsetName)
+		}
+		charset = charset.Subtract(charsetToExclude)
+		for i, includedCharset := range includedCharsets {
+			if charsetToExclude.Equals(includedCharset) {
+				return ErrFlagsMutuallyExclusive(cmd.includes[i], charsetName)
+			}
+		}
+	}
+
+	var options []randchar.Option
+	for _, minFlag := range cmd.mins {
+		charset, count, err := parseMinFlag(minFlag)
+		if err != nil {
+			return err
+		}
+		options = append(options, randchar.Min(count, charset))
+	}
+
+	if charset.Size() <= 0 {
+		return ErrCharsetSizeNonPositive
+	}
```

Comment:
We discussed moving this to `ranchar.NewRand`, as this check should be done on every call of NewRand.

Reply:
This state is no longer reachable with the new `--charset` flag. But I agree this check should be done in the client.